System stats - API addition

deepikabhavnani opened the pull request:
### Description

System Stats Addition. ~~Dependent on previous PR's. (Commits valid post ec71e4641cd24b4d9993ebf9f487cbe24d014441)~~

Status : ~~Work in progress~~ Done

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] New target
    [X] Feature
    [ ] Breaking change

Commit:
System Info API addition

API added to get the system information like CPU ID, compiler ID and
compiler version.

cmonr commented:
@deepikabhavnani Is this blocked on #508?

deepikabhavnani commented:
@cmonr - This is not dependent on "Thread/System API addition #508" which is on Handbook. But should be merged post https://github.com/ARMmbed/mbed-os/pull/6795

Commit:
Added test for mbed_stats_sys_get() API

deepikabhavnani commented:
Rebased on top of master.. No PR dependency now.

deepikabhavnani commented:
@SenRamakri - Please review

0xc0170 commented:
/morph build

deepikabhavnani commented:
@0xc0170 - Command used in build does not contain ALL flags enabled. Will have to re-run once @studavekar does that change.

studavekar commented:
> @0xc0170 - Command used in build does not contain ALL flags enabled. Will have to re-run once @studavekar does that change.

We should be good to run trigger the build on this PR, Logs from one of the PR 

https://mbed-os.s3-eu-west-1.amazonaws.com/builds/6832/PASS/DISCO_F303VC/GCC_ARM/6154fd259845423b63fd8cc6a06f4798f3dd115e_build_log_DISCO_F303VC_GCC_ARM.txt

new command line looks like `mbed test --compile -m <target> -t <toolchain> -DMBED_HEAP_STATS_ENABLED=1 -DMBED_STACK_STATS_ENABLED=1 -DMBED_TRAP_ERRORS_ENABLED=1 -DMBED_ALL_STATS_ENABLED`

cmonr commented:
/morph build

Rerunning to use new build parameters.

bulislaw reviewed (dismissed):
One thing I would like to add to the stats is the Mbed OS version.

0xc0170 commented:
mbed OS version +1

to consider to be having in system stats if we have there mbed OS version - https://github.com/ARMmbed/mbed-os/issues/6897 - having sha available or dirty flag, this might not be related to this one?

deepikabhavnani commented:
@bulislaw Will add Mbed OS version.

@0xc0170 - I am not sure if we can add SHA, but can look upon that in future.

Commit:
OS version added + CPUID for cortex-M only

CPUID base register is available for Cortex-M processors only.
Cortex-A devices have Main ID Register, which can be used in future to
get processor info.

deepikabhavnani commented:
Rebased, added Mbed  OS version and resolved Cortex-A compilation issue.

deepikabhavnani commented:
Testing build to confirm Cortex-A issues resolved.

/morph build

cmonr commented:
/morph test

cmonr commented:
@0xc0170 @SenRamakri @bulislaw Are y'all alright with the updated changes?
